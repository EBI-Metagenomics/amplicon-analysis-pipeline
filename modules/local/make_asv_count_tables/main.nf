
process MAKE_ASV_COUNT_TABLES {

    label 'light'
    // publishDir "${outdir}/${project}/${sampleId}/asv-gen/${var_region}", mode : "copy" 

    input:
    tuple val(meta), path(maps), path(taxa), path(extracted_var_path), path(reads)

    output:
    tuple val(meta), path("*asv_krona_counts.txt"), emit: asv_count_tables_out

    """
    if [[ ${meta.single_end} = true ]]; then
        zcat $reads | sed -n "1~4p" > headers.txt
        python /hps/software/users/rdf/metagenomics/service-team/users/chrisata/asv_gen/bin/make_asv_count_table.py -t $taxa -f $maps -a $extracted_var_path -hd ./headers.txt  -s ${meta.id}
    else
        zcat ${reads[0]} | sed -n "1~4p" > headers.txt
        python /hps/software/users/rdf/metagenomics/service-team/users/chrisata/asv_gen/bin/make_asv_count_table.py -t $taxa -f ${maps[0]} -r ${maps[1]} -a $extracted_var_path -hd ./headers.txt  -s ${meta.id}
    fi    
    """

}
