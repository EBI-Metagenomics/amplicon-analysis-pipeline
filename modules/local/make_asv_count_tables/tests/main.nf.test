nextflow_process {

    name "Test Process MAKE_ASV_COUNT_TABLES"
    script "../main.nf"
    process "MAKE_ASV_COUNT_TABLES"
    tag "modules"
    tag "make_asv_count_tables"

    test("Should run without failures") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    "16S-V3-V4",
                    [
                        file('${projectDir}/modules/local/make_asv_count_tables/tests/data/test_1_map.txt', checkIfExists: true),
                        file('${projectDir}/modules/local/make_asv_count_tables/tests/data/test_2_map.txt', checkIfExists: true) 
                    ],
                    file('${projectDir}/modules/local/make_asv_count_tables/tests/data/test_proportion_chimeric.txt', checkIfExists: true),
                    file('${projectDir}/modules/local/make_asv_count_tables/tests/data/test_proportion_matched.txt', checkIfExists: true),
                    file('${projectDir}/modules/local/make_asv_count_tables/tests/data/test_taxa.tsv', checkIfExists: true),
                    file('${projectDir}/modules/local/make_asv_count_tables/tests/data/test.16S.V3-V4.txt', checkIfExists: true),
                    [ 
                        file('${projectDir}/modules/local/make_asv_count_tables/tests/data/test_1.fastp.fastq.gz', checkIfExists: true),
                        file('${projectDir}/modules/local/make_asv_count_tables/tests/data/test_2.fastp.fastq.gz', checkIfExists: true) 
                    ]
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.asv_count_tables_out).match("asv_count_tables_out") }
            )
        }

    }

}
