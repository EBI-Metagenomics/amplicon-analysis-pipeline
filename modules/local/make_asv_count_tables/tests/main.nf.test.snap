{
    "asv_count_tables_out": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "16S-V3-V4",
                    "test_16S.V3-V4_asv_krona_counts.txt:md5,d9bad313abd84329ac187ced4b92857e"
                ]
            ]
        ],
        "timestamp": "2023-11-08T11:27:57.725803699"
    }
}